tests/testthat/test-type-sf.R

```diff
@@ -0,0 +1,84 @@
+
+skip_if_not_installed("sf")
+
+# Avoids adding `sf` to Suggests
+import_from("sf", c(
+  "st_sfc",
+  "st_point",
+  "st_bbox",
+  "st_precision",
+  "st_crs",
+  "st_linestring"
+))
+
+
+# These tests are missing upstream
+
+test_that("can cast `sfc` vectors", {
+  pt <- st_sfc(st_point())
+  ln <- st_sfc(st_linestring())
+  expect_identical(class(vctrs::vec_cast(pt, ln)), c("sfc_LINESTRING", "sfc"))
+})
+
+
+# Keep these tests in sync with
+# https://github.com/r-spatial/sf/blob/master/tests/testthat/test_vctrs.R
+
+test_that("`sfc` vectors are treated as vectors", {
+	expect_true(vctrs::vec_is(st_sfc(st_point())))
+})
+
+test_that("`n_empty` attribute of `sfc` vectors is restored", {
+	pt1 = st_sfc(st_point(c(NA_real_, NA_real_)))
```

Comment:
Looks like the indenting is off here?

Reply:
yup the sf codebase uses tabs. Since this is imported from upstream it's better not to make any changes to keep the diffs clean.